include/git2/notes.h

```diff
@@ -223,6 +223,27 @@ GIT_EXTERN(int) git_note_remove(
 	const git_oid *oid);
 
 /**
+ * Remove the note for an object
+ *
+ * @param notes_commit_out pointer to store the commit (optional);
+ *					NULL in case of error
```

Comment:
Which commit is this exactly? Is this any different from `notes_commit`? I think this concept should be explained a bit better.

Reply:
Yes it should be, when a note is removed we make a new tree with all notes sans the note to be removed, so a new commit is created pointing to this new tree. Even if we remove the last note, we still have a pointer (the commit) to an empty tree object.